src/lib/utils/tokens.ts

```diff
@@ -38,27 +31,15 @@ import {
   WSTETH,
 } from '@/constants/tokens'
 
-import { isSameAddress } from '.'
-
 export function getAddressForToken(
   token: Token,
   chainId: number | undefined,
 ): string | undefined {
-  if (token.symbol === IndexToken.symbol) return token.address
-  switch (chainId) {
-    case ARBITRUM.chainId:
-      return token.arbitrumAddress
-    case BASE.chainId:
-      return token.baseAddress
-    case MAINNET.chainId:
-      return token.address
-    case OPTIMISM.chainId:
-      return token.optimismAddress
-    case POLYGON.chainId:
-      return token.polygonAddress
-    default:
-      return undefined
-  }
+  const nativeToken = getNativeToken(chainId)
```

Comment:
we could probably extend the tokenLists with some native token functionality, but otherwise lgtm